accord-core/src/test/java/accord/impl/basic/SimulatedDelayedExecutorService.java

```diff
@@ -26,12 +31,105 @@
 import accord.burn.random.RandomWalkRange;
 import accord.utils.RandomSource;
 
-public class SimulatedDelayedExecutorService extends TaskExecutorService
+import static java.util.concurrent.TimeUnit.NANOSECONDS;
+
+public class SimulatedDelayedExecutorService extends TaskExecutorService implements ScheduledExecutorService
 {
+    private class ScheduledTask<T> extends Task<T> implements ScheduledFuture<T>
+    {
+        private final long sequenceNumber;
+        private final long periodMillis;
+        private long timeMillis;
+        private ScheduledTask(long sequenceNumber, long value, TimeUnit unit, Callable<T> fn)
+        {
+            super(fn);
+            this.sequenceNumber = sequenceNumber;
+            periodMillis = unit.toMillis(value);
+        }
+
+        @Override
+        public long getDelay(TimeUnit unit)
+        {
+            return unit.convert(timeMillis - pending.nowInMillis(), TimeUnit.MILLISECONDS);
+        }
+
+        @Override
+        public int compareTo(Delayed other)
+        {
+            if (other == this) // compare zero if same object
+                return 0;
+            if (other instanceof ScheduledTask) {
+                ScheduledTask<?> x = (ScheduledTask<?>)other;
+                long diff = timeMillis - x.timeMillis;
+                if (diff < 0)
+                    return -1;
+                else if (diff > 0)
+                    return 1;
+                else if (sequenceNumber < x.sequenceNumber)
+                    return -1;
+                else
+                    return 1;
+            }
+            long diff = getDelay(NANOSECONDS) - other.getDelay(NANOSECONDS);
+            return (diff < 0) ? -1 : (diff > 0) ? 1 : 0;
+        }
+
+        @Override
+        public void run()
+        {
+            boolean periodic = periodMillis != 0;
+            if (!periodic)
+            {
+                super.run();
+            }
+            else
+            {
+                // run without setting the result
+                try
+                {
+                    fn.call();
+                    long nowMillis = pending.nowInMillis();
+                    if (periodMillis < 0)
+                    {
+                        // scheduleWithFixedDelay
+                        timeMillis = nowMillis + (-periodMillis);
+                    }
+                    else
+                    {
+                        // scheduleAtFixedRate
+                        timeMillis = timeMillis + periodMillis;
+                    }
+                    long delayMillis = timeMillis - nowMillis;
+                    if (delayMillis < 0)
+                        delayMillis = 0;
+                    schedule(this, delayMillis, TimeUnit.MILLISECONDS);
+                }
+                catch (Throwable t)
+                {
+                    // setSuccess may trigger callbacks, which may throw exceptions... if that happens then the result was already set
```

Comment:
this is not true

Reply:
`setSuccess` does end up calling `notify` and the listeners can all throw so we can get an exception here that is from the callbacks not the actual task in which case `setFailure` will think we attempted to `setFailure` on something already completed and throw illegal state exception.

In this case no result is ever set so no point in attempting `tryFaiure`, but in the parent it does make sense to use `tryFailure` so we can tell which occurred.